library_generation/DEVELOPMENT.md

```diff
@@ -48,32 +48,36 @@ Although the scripts are designed to be run in a Docker container, you can also
 run them directly. This section explains how to run the entrypoint script
 (`library_generation/cli/entry_point.py`).
 
-## Installing prerequisites
+## Assumptions made by the scripts
+### The Hermetic Build's well-known folder
+Located in `${HOME}/.library_generation`, this folder is assumed by the scripts
+to contain the generator JAR. Please note that this is a recent feature and only
+this jar is expected to be there. Soon enough, more binaries such as the gRPC
+and protoc plugins will also be stored here. Developers must make sure
+this folder is properly configured before running the scripts locally.
+Note that this relies on the `HOME` en var which is always
+defined as per
+[POSIX env var definition](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html).
 
-In order to run the generation scripts directly, there are a few tools we
-need to install beforehand.
+#### Put the gapic-generator-java jar in its well-known location
 
-### Install synthtool
+Run `cd sdk-platform-java && mvn install -DskipTests -Dclirr.skip
+-Dcheckstyle.skip`. This will generate a jar located in
+`~/.m2/repository/com/google/api/gapic-generator-java/{version}/gapic-generator-java-{version}.jar`
 
-It requires python 3.x to be installed.
-You will need to specify a committish of the synthtool repo in order to have
-your generation results matching exactly what the docker image would produce.
-You can achieve this by inspecting `SYNTHTOOL_COMMITISH` in
-`.cloudbuild/library_generation/library_generation.Dockerfile`.
+Then `mv` the jar into the well-known location of the jar. The
```

Comment:
When building an image, the generator jar is already put into the well-known location.

Why does a developer need to build and move the jar again?

Reply:
They wouldn't need to prepare this jar when running the Docker image since the Dockerfile will take care of it. The thing is that this new section is meant for local development (it is under that section).